inference-engine/src/gna_plugin/gna_model_serial.cpp

```diff
@@ -336,15 +332,20 @@ void GNAModelSerial::Export(void * basePointer, size_t gnaGraphSize, std::ostrea
     writeBits(header, os);
 
     for (auto &name : inputNames) {
-        writeBits(name.size(), os);
+        ulong test = sizeof(size_t);
+        writeBits(sizeof(size_t), os);
+        writeBits(name.size() + 1, os);
         writeNBytes(name.c_str(), name.size(), os);
+        writeBits('\0', os);
     }
     for (const auto &input : inputs) {
         writeBits(convert_to_serial(input), os);
     }
     for (auto &name : outputNames) {
-        writeBits(name.size(), os);
+        writeBits(sizeof(size_t), os);
+        writeBits(name.size() + 1, os);
```

Comment:
Don't add `writeBits(sizeof(size_t), os);`
it does not remove ambiguity it adds the field but the ambiguity remains.
To remove ambiguity related to the size of size_t you can use static_cast<>():
`writeBits(static_cast<uint64_t>(sizeof(name.size() + 1)), os);`